File: RB101_Programming_Foundations/Small Probems/Easy4/Easy4_Python/convert_string_to_number.py
'''
input : string of numbers
output : integers of original string

Algo :
    create an array of strings from 0 to 9 ['0', '1', ... '9']
    create a variable for the number, starts with 0
    iterate through each character of the string
        - convert the current char to number by finding its index in the array
        - multiplies the original number by 10 and add this number to it
    return the number
'''

def string_to_integer(string):
    number_array = [str(x) for x in range(0, 10)]

    number = 0

    for chr in string:
        number = number * 10 + number_array.index(chr)
    return number



print(string_to_integer('4321') == 4321)
print(string_to_integer('570') == 570)

File: js210_fundamentals_of_JS_Programming/problem_sets/javascript_basics/convert_string_to_number.js
const DIGITS = {
  0: 0,
  1: 1,
  2: 2,
  3: 3,
  4: 4,
  5: 5,
  6: 6,
  7: 7,
  8: 8,
  9: 9,
};

function stringToInteger(string) {
  let number = 0;

  for (let index = 0, len = string.length; index < len; index++) {
    let chr = string[index];
    number = number * 10 + DIGITS[chr];
  }
  return number;
}

console.log(stringToInteger('4321'));
console.log(stringToInteger('570'));
console.log(stringToInteger('100')); // 100
